### DISCUSSION TITLE
How to use an external Python function in Comsol

### DISCUSSION THREAD

**Discussion by shikhar58** [MPh-py/MPh#95] [General]
I have been using MPh and it is working fine. In one case, I have a discrete values for temperature at different spatial points x. I want to use a function T(x) based on the interpolation of the discrete values, and use this function as my initial condition of temperature in the model.

I have been using an interpolation function from scipy, but when I feed it into comsol model through MPh it does not recognise the function. Please let me know how to deal with it

Thanks in advance.

**Comment by john-hen** [comment]
Hi. MPh is a scripting interface *for* Comsol. It lets you automate the things that Comsol does, but it does not teach Comsol new tricks. I don't think Comsol supports calling external Python functions, but that's really a Comsol question, and not about MPh.

**Reply by shikhar58** [reply]
That was helpful. Thank you
